### PR TITLE
fix(psalm): adapt to updated OCP dev-master return types

### PR DESCRIPTION
**Pull request by GretaD**
Tests are failing here: https://github.com/nextcloud/mail/pull/13018
Tried to fix it with claude

- IRequest::getUploadedFile now declared as array|null; use inline isset() so psalm can narrow the type correctly
- IUser::getUID now declared as non-nullable string; remove redundant null guard in Provisioning::buildUserEmail

Assisted-by: ClaudeCode:claude-sonnet-4-6

## 🤖 AI (if applicable)

- [x] The content of this PR was partly or fully generated using AI

## Summary by CodeRabbit

* **Refactor**
  * Optimized internal attachment handling logic.
  * Simplified user identification placeholder processing.

### CODE REVIEW

File: lib/Controller/MessageApiController.php [hunk 1]
```diff
@@ -455,8 +455,7 @@ public function getAttachment(int $id, string $attachmentId): DataResponse {
 	 */
 	private function handleAttachments(): array {
 		$fileAttachments = $this->request->getUploadedFile('attachments');
-		$hasAttachments = isset($fileAttachments['name']);
-		if (!$hasAttachments) {
+		if (!isset($fileAttachments['name'])) {
```
Comment (ChristophWurst) [inline lib/Controller/MessageApiController.php:458]: I don't get this one. It's the identical code, just inlined?

**Comment by ChristophWurst** [comment]
@coderabbitai review

File: lib/Controller/MessageApiController.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/Controller/MessageApiController.php:458]: It's because Psalm has issues detecting the types otherwise.

**Review by ChristophWurst — APPROVED** [review]
Pushing a commit to also fix the OOO controller test that fails now because the iuser mock returns an empty string instead of null
